test/unit/supervisor/kubernetes-api-wrappers.test.ts

```diff
@@ -46,6 +46,84 @@ tap.test('calculateSleepSeconds', async (t) => {
   );
 });
 
+tap.test('retryKubernetesApiRequest for ECONNREFUSED error', async (t) => {
+  const retryableErrorResponse = {code: 'ECONNREFUSED'};
+
+  t.rejects(
```

Comment:
I think we have to await here, otherwise `t.rejects()` creates a Promise that will run in the background, this can make the tests flaky

Reply:
I can do it. Just confirm, because I copy this behaviour from this test. I think I should include `await` for all of them too, right?

```ts
tap.test('retryKubernetesApiRequest for retryable errors', async (t) => {
  const retryableErrorResponse = {response: {statusCode: 429}};

  t.rejects(
    () => kubernetesApiWrappers.retryKubernetesApiRequest(() => Promise.reject(retryableErrorResponse)),
    'eventually throws on repeated retryable error responses',
  );

  let failures = 0;
  const functionThatFailsJustEnoughTimes = () => {
    if (failures < kubernetesApiWrappers.ATTEMPTS_MAX - 1) {
      failures +=1;
      return Promise.reject(retryableErrorResponse);
    }
    return Promise.resolve('egg');
  };

  const successfulResponse = await kubernetesApiWrappers.retryKubernetesApiRequest(functionThatFailsJustEnoughTimes);
  t.equals(
    successfulResponse,
    'egg',
    'keeps retrying on 429 as long as we don\'t cross max attempts',
  );

  failures = 0;
  const functionThatFailsOneTooManyTimes = () => {
    if (failures < kubernetesApiWrappers.ATTEMPTS_MAX) {
      failures +=1;
      return Promise.reject(retryableErrorResponse);
    }
    return Promise.resolve('egg');
  };

  t.rejects(
    () => kubernetesApiWrappers.retryKubernetesApiRequest(functionThatFailsOneTooManyTimes),
    'failure more than the maximum, rejects, even for a retryable error response',
  );
});
```

Reply:
Yes for sure, but my suggestion would be the improvement to come as a separate PR! 🙏 

Here's an example in another repo where we await: https://github.com/snyk/snyk-docker-plugin/blob/8f1d1e9e4a91fb642bf038d117ff548ea1a3b532/test/system/experimental.test.ts#L63-L71